fhir-term/src/main/java/com/ibm/fhir/term/util/ConceptMapSupport.java

```diff
@@ -11,30 +11,68 @@
 import java.util.stream.Collectors;
 
 import com.ibm.fhir.model.resource.ConceptMap;
-import com.ibm.fhir.model.resource.ConceptMap.Group.Element.Target;
+import com.ibm.fhir.model.resource.ConceptMap.Group;
+import com.ibm.fhir.model.resource.ConceptMap.Group.Element;
 import com.ibm.fhir.model.type.Coding;
+import com.ibm.fhir.model.type.String;
+import com.ibm.fhir.model.type.Uri;
 import com.ibm.fhir.registry.FHIRRegistry;
+import com.ibm.fhir.term.spi.Match;
 
 public final class ConceptMapSupport {
     private ConceptMapSupport() { }
 
-    public static List<Target> translate(ConceptMap conceptMap, Coding coding) {
+    public static List<Match> translate(ConceptMap conceptMap, Coding coding) {
         return conceptMap.getGroup().stream()
                 .filter(group -> group.getSource() != null)
                 .filter(group -> group.getSource().equals(coding.getSystem()))
                 .filter(group -> group.getSourceVersion() == null ||
                     coding.getVersion() == null ||
                     group.getSourceVersion().equals(coding.getVersion()))
-                .map(group -> group.getElement())
-                .flatMap(List::stream)
-                .filter(element -> element.getCode() != null)
-                .filter(element -> element.getCode().equals(coding.getCode()))
-                .map(element -> element.getTarget())
+                .map(group -> group.toBuilder()
+                    .element(group.getElement().stream()
+                        .filter(element -> element.getCode() != null)
+                        .filter(element -> element.getCode().equals(coding.getCode()))
```

Comment:
just noticing this now... any reason to combine 34,35

Reply:
I keep them separate for readability